src/dbus-function-ifaces/gadgetd-serial-function-iface.c

```diff
@@ -109,7 +114,15 @@ function_serial_attrs_get_property(GObject    *object,
 	FunctionSerialAttrs *serial_attrs = FUNCTION_SERIAL_ATTRS(object);
 	GadgetdFunctionObject *function_object;
 	usbg_function *f;
-	usbg_function_attrs f_attrs;
+	/* FIXME: find a better solution to a union here? */
+	union {
+		struct usbg_f_net_attrs net;
+		char *ffs_dev_name;
+		struct usbg_f_ms_attrs ms;
+		struct usbg_f_midi_attrs midi;
+		int serial_port_num;
+		char *phonet_ifname;
+	} f_attrs;
```

Comment:
Yes there is a better solution.

libusbgx provides now specialized getters for each function types. So instead of using this union  and a generic usbg_get_function_attrs() you may simply use usbg_f_serial_get_port_num(). See https://github.com/libusbgx/libusbgx/blob/master/include/usbg/function/serial.h for more details